File: room-demo-android-q/src/main/java/com/example/pavneet_singh/room_demo_android_q/notedb/dao/NoteDao.java
package com.example.pavneet_singh.room_demo_android_q.notedb.dao;


import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.example.pavneet_singh.room_demo_android_q.notedb.model.Note;
import com.example.pavneet_singh.room_demo_android_q.util.Constants;

import java.util.List;

/**
 * Created by Pavneet_Singh on 12/31/17.
 */

@Dao
public interface NoteDao {

    @Query("SELECT * FROM " + Constants.TABLE_NAME_NOTE)
    List<Note> getNotes();

    /*
     * Insert the object in database
     * @param note, object to be inserted
     */
    @Insert
    long insertNote(Note note);

    /*
     * update the object in database
     * @param note, object to be updated
     */
    @Update
    void updateNote(Note repos);

    /*
     * delete the object from database
     * @param note, object to be deleted
     */
    @Delete
    void deleteNote(Note note);

    // Note... is varargs, here note is an array
    /*
     * delete list of objects from database
     * @param note, array of oject to be deleted
     */
    @Delete
    void deleteNotes(Note... note);

}

File: room-demo-kotlin-mvvm-dagger/src/main/java/com/example/pavneet_singh/room_demo_kotin_mvvm_dagger/notedb/dao/NoteDao.kt
package com.example.pavneet_singh.room_demo_kotin_mvvm_dagger.notedb.dao

import androidx.lifecycle.LiveData
import androidx.room.*
import com.example.pavneet_singh.room_demo_kotin_mvvm_dagger.notedb.model.Note
import com.example.pavneet_singh.room_demo_kotin_mvvm_dagger.util.Constants

/**
 * Created by Pavneet_Singh on 2020-01-25.
 */

@Dao
interface NoteDao {

    /**
     * Room will take care of managing execution on background thread with LiveData
     * Therefore no need of suspend and coroutines
     */
    @Query("SELECT * FROM " + Constants.TABLE_NAME_NOTE)
    fun getNotes(): LiveData<List<Note>>

    /*
     * Insert the object in database
     * @param note, object to be inserted
     */
    @Insert
    suspend fun insertNote(note: Note): Long

    /*
     * update the object in database
     * @param note, object to be updated
     */
    @Update
    suspend fun updateNote(repos: Note): Int

    /*
     * delete the object from database
     * @param note, object to be deleted
     */
    @Delete
    suspend fun deleteNote(note: Note): Int

    // Note... is varargs, here note is an array
/*
     * delete list of objects from database
     * @param note, array of oject to be deleted
     */
    @Delete
    suspend fun deleteNotes(vararg note: Note): Int
}
